DSDEGP-2829: add option to control metrics delivery for deliver cram command

pshapiro4broad opened the pull request:
Here's another approach to fix this issue that only affects the cram delivery command.

----

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [x] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

pshapiro4broad committed:
DSDEGP-2829: add option to control metrics delivery for deliver cram command

tlangs reviewed (approved):
Looks good, but we should probably put in some integration tests to verify functionality from the command line. You can pull those test from my branch.

tlangs commented on clio-client/src/main/scala/org/broadinstitute/clio/client/metadata/CramDeliverer.scala:
```diff
@@ -37,7 +37,7 @@ class CramDeliverer extends MetadataMover[CramMetadata] {
     ).flatten
 
-    if (src.regulatoryDesignation.exists(_.equals(RegulatoryDesignation.ResearchOnly))) {
+    if (deliverSampleMetrics) {
```
We should talk about where we want to the check for regulatory designation to happen. Doing it in the delivery WDL will be kinda unwieldy. We don't have the Regulatory Designation available on a per-sample basis when we call `MoveSamples` in the WDL to make the decision of whether or not to actually move sample level metrics.

als364 replied:
Why not just keep it here?

pshapiro4broad replied:
The original JIRA says
> All sample-level metrics files should be delivered for most samples but not for clinical samples or for fee-for-service projects

Since we don't have a way to test for fee-for-service (and since we don't know yet how users will request this delivery feature) I thought it best to only use the command line argument to toggle this setting.

tlangs replied:
We know fee-for-service when delivery is started, so that flag can be set easily in the WDL. We don't know the Regulatory Designation of each sample in the WDL though, and its much more effort to get that into the WDL than to just check for it when we're already in Clio.

pshapiro4broad replied:
OK, I can add it back in. The original issue says `not for clinical samples` but the code you added tests for `ResearchOnly`. Is there a reason why you didn't use `!Clinical`?

I don't see why anyone would need to know regulatory designation, though. We aren't doing to deliver these metrics unless someone runs delivery with a command line option. This is a manual step; at that point all the checks should've been done to confirm that this is the right thing to do. If we get to this point and silently omit the metrics the user has asked for, that doesn't seem correct either. I guess we could add code to fail in that case. But now we're adding extra code and we don't yet know when this feature will be used.

als364 commented on clio-client/src/test/scala/org/broadinstitute/clio/client/metadata/CramDelivererSpec.scala:
```diff
@@ -51,10 +51,12 @@ class CramDelivererSpec extends FlatSpec with Matchers {
+  private val delivererNoMetrics = CramDeliverer(deliverSampleMetrics = false)
+
+  it should "generate ops to move the cram & crai, write the cram md5 and copy metrics files" in {
+    val (delivered, ops) = delivererWithMetrics.moveInto(metadata, destination)
```
Please add a check to verify that the metrics files are in the right place

pshapiro4broad replied:
This is already done below, see the lines regarding `preAdapterSummaryMetricsPath`, `preAdapterDetailMetricsPath`, etc.

als364 replied:
github autocollapse strikes again.

als364 commented on clio-client/src/test/scala/org/broadinstitute/clio/client/metadata/CramDelivererSpec.scala:
```diff
@@ -109,10 +111,11 @@ class CramDelivererSpec extends FlatSpec with Matchers {
     val basename = "the-new-basename"
 
-    val (delivered, ops) = deliverer.moveInto(metadata, destination, Some(basename))
+    val (delivered, ops) =
```
Please add a check to verify that the metrics files are in the right place

pshapiro4broad committed:
DSDEGP-2829: add option to control metrics delivery for deliver cram command

- if user requests sample level metrics, and the sample is clinical, throw an error
- add integration tests to check all combinations of sample delivery and regulatory designation
- add isClinical to RegulatoryDesignation

pshapiro4broad committed:
Merge branch 'develop' into ps_DSDEGP-2829_add_metrics_delivery_option

# Conflicts:
#	clio-client/src/main/scala/org/broadinstitute/clio/client/metadata/CramDeliverer.scala
